ed_sensor_integration/tools/segmenter.cpp

```diff
@@ -25,251 +22,129 @@
-        std::cerr << error.str() << std::endl;
-        return false;
+        cv::Point d(2, 2);
+        cv::rectangle(canvas, f * bb_min, f * bb_max, cv::Scalar(255, 255, 255), 2);
+        cv::rectangle(canvas, f * bb_min - d, f * bb_max + d, cv::Scalar(0, 0, 0), 2);
+        cv::rectangle(canvas, f * bb_min + d, f * bb_max - d, cv::Scalar(0, 0, 0), 2);
     }
-
-    // Reset world
-    world_model = ed::WorldModel();
-
-    // Update world
-    world_model.update(req);
-
-    return true;
 }
 
-// ----------------------------------------------------------------------------------------------------
-
+/**
+ * @brief usage: instruct user how to use the executable
+ */
 void usage()
 {
-    std::cout << "Usage: ed_segmenter IMAGE-FILE-OR-DIRECTORY" << std::endl;
+    std::cout << "Usage: ed_segmenter IMAGE-FILE-OR-DIRECTORY [WORLDMODEL-NAME] [ENTITY-ID]" << std::endl;
 }
 
-// ----------------------------------------------------------------------------------------------------
 
 int main(int argc, char **argv)
 {
-    ros::init(argc, argv, "ed_segmenter");
-    ros::NodeHandle nh;
-
-    if (argc != 2)
+    if (argc != 4)
     {
-        usage();
-        return 1;
+        if (argc == 2)
+        {
+            std::cout << "No worldmodel provided! Using empty worldmodel and no segmentation area" << std::endl;
+        }
+        else
+        {
+            usage();
+            return 1;
+        }
     }
 
-//    std::string model_name = argv[2];
-
-//    ed::WorldModel world_model;
-//    if (!loadWorldModel(model_name, world_model))
-//        return 1;
-
     tue::filesystem::Path path = argv[1];
     if (!path.exists())
     {
         std::cerr << "Path '" << path << "' does not exist." << std::endl;
         return 1;
     }
 
-    tue::filesystem::Crawler crawler;
+    std::string model_name;
+    ed::WorldModel world_model;
+    std::string entity_id;
+    ed::EntityConstPtr e;
+    std::string area_description;
 
-    if (path.isDirectory())
-        crawler.setRootPath(path);
-    else
-        crawler.setRootPath(path.parentPath());
+    if (argc == 4)
+    {
+        model_name = argv[2];
+
+        if (!ed_sensor_integration::loadWorldModel(model_name, world_model))
+        {
+            std::cerr << "World model '" << model_name << "' could not be loaded." << std::endl;
+            return 1;
+        }
+
+        entity_id = argv[3];
+        area_description = "on_top_of " + entity_id;
+        e = world_model.getEntity(entity_id);
+        if (!e)
+        {
+            std::cerr << "Entity '" << entity_id << "' could not be found in world model '" << model_name << "'." << std::endl;
+            return 1;
+        }
+    }
+
+    ed_sensor_integration::SnapshotCrawler crawler(path);
 
     Updater updater;
 
-    std::vector<Snapshot> snapshots;
+    std::vector<ed_sensor_integration::Snapshot> snapshots;
     unsigned int i_snapshot = 0;
 
-    while(ros::ok())
+    while(true)
```

Comment:
Why? I think `ros::ok()` is better.

Reply:
Because this executable does nothing with ros.